adagrams/game.py

```diff
@@ -1,13 +1,138 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_VALUES = {
+    'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+    'D': 2, 'G': 2,
+    'B': 3, 'C': 3, 'M': 3, 'P': 3,
+    'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+    'K': 5,
+    'J': 8, 'X': 8,
+    'Q': 10, 'Z': 10
+}
 def draw_letters():
-    pass
+    """
+    build a hand of 10 letters for the user
+    Parameters:    None
+    Returns:
+    a list of strings with single letters
+    """     
+    #step1: create a copy of the LETTER_POOL and save all keys in a list
+    letter_count_dict = {}
+    letter_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_count_dict[letter] = count
+        letter_list.append(letter)
+    #step2:randomly pick 10 letters from the letter_pool: for each pick, update count, remove letter from letter_list if count gets to 0;  
+    hand = []
+    while len(hand) < 10:
+        index = randint(0, len(letter_list) - 1)
+        letter = letter_list[index]
+        count = letter_count_dict[letter] 
+        hand.append(letter)
+        letter_count_dict[letter] = count - 1
+        # new_count = count  - 1
+        if count == 1:
+            letter_list.pop(index)
```

Comment:
As mentioned before, using the dictionary doesn't solve the problem of distribution. Right now, you are picking a random index from the list `['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L' 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']` which gives every letter an equal chance of being chosen, even with decrementing the count in the dictionary. If we had instead used LETTER_POOL to build a pile of letters, we could destroy that pile by removing letters as we randomly pick them. 

This is not something that requires fixing, but you are welcome to try rewriting this function if you would like!